dev/Bootstrap/CS/Microsoft.WindowsAppSDK.Bootstrap.Net/Bootstrap.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Windows.ApplicationModel.DynamicDependency
+{
+    // The version of an MSIX package. This is logically `Major.Minor.Build.Revision` and can be expressed as...
+    // * individual `ushort` values (uint16)
+    // * an unsigned `ulong` value (uint64)
+    // * a dot-string notation ("major.minor.build.revision")
+    public struct PackageVersion
+    {
+        public ushort Major;
+        public ushort Minor;
+        public ushort Build;
+        public ushort Revision;
+
+        // Create an instance with the value `major.0.0.0`.
+        public PackageVersion(ushort major) :
+            this(major, 0, 0, 0)
+        {
+        }
+
+        // Create an instance with the value `major.minor.0.0`.
+        public PackageVersion(ushort major, ushort minor) :
+            this(major, minor, 0, 0)
+        {
+        }
+
+        // Create an instance with the value `major.minor.build.0`.
+        public PackageVersion(ushort major, ushort minor, ushort build) :
+            this(major, minor, build, 0)
+        {
+        }
+
+        // Create an instance with the value `major.minor.build.revision`.
+        public PackageVersion(ushort major, ushort minor, ushort build, ushort revision)
+        {
+            Major = major;
+            Minor = minor;
+            Build = build;
+            Revision = revision;
+        }
+
+        // Create an instance from a version as a uint64.
+        public PackageVersion(ulong version) :
+            this((ushort)(version >> 48), (ushort)(version >> 32), (ushort)(version >> 16), (ushort)version)
+        {
+        }
+
+        // Return the version as a uint64.
+        public ulong ToVersion()
+        {
+            return (((ulong)Major) << 48) | (((ulong)Minor) << 32) | (((ulong)Build) << 16) | ((ulong)Revision);
+        }
+
+        // Return the string as a formatted value "major.minor.build.revision".
+        public override string ToString()
+        {
+            return $"{Major}.{Minor}.{Build}.{Revision}";
+        }
+    };
+
+    internal static class NativeMethods
+    {
+        [DllImport("Microsoft.WindowsAppSDK.Bootstrap.dll", EntryPoint = "MddBootstralInitialize", CharSet = CharSet.Unicode, ExactSpelling = true, PreserveSig = false)]
```

Comment:
Spelling error on EntryPoint. Is there a test for this?

Reply:
I thought so too but "Entrypoint" doesn't compile. "EntryPoint" (cap P) is right - per VS compiling the project and [docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportattribute.entrypoint?view=net-5.0)

Reply:
I think he was referring to `MddBootstralInitialize`, note the missing `p` in bootstrap.

Reply:
Doh!